packages/webhooks/model-validator/validator.js

```diff
@@ -0,0 +1,208 @@
+const fs = require('fs')
+const tjs = require('typescript-json-schema')
+const Ajv = require('ajv')
+const get = require('lodash/get')
+
+/**
+ * Consumes path of generated schema.ts file and creates schema-generator from it
+ * Validates the correctness of the requested fields and filters for registered models
+ */
+class WebHookModelValidator {
+    constructor (schemaPath) {
+        if (typeof schemaPath !== 'string' || schemaPath.split('.').pop() !== 'ts') {
+            throw new Error('Schema file type was not supported. For now only .ts files are supported.')
+        } else if (!fs.existsSync(schemaPath)) {
+            throw new Error('Schema file does not exist!')
+        }
+        const program = tjs.getProgramFromFiles([schemaPath])
+        const settings = {
+            noExtraProps: true,
+        }
+        this.generator = tjs.buildGenerator(program, settings)
+        this.models = []
+        this.filterValidators = {}
+        this.fieldsValidators = {}
+    }
+
+    /**
+     * Registers GraphQL model from schema in internal base
+     * Also generates JSON-schema validators for validating its fields and filters
+     * @param {string} modelName
+     */
+    registerModel (modelName) {
+        if (this.models.includes(modelName)) return
+
+        this.models.push(modelName)
+
+        const modelFilterSchema = this.generator.getSchemaForSymbol(`${modelName}WhereInput`)
+        const ajv = new Ajv({ strict: true, allErrors: true })
+        ajv.validateSchema(modelFilterSchema)
+        if (ajv.errors) {
+            throw new Error(`Schema for class ${modelName}WhereInput was not valid!`)
+        }
+        this.filterValidators[modelName] = ajv.compile(modelFilterSchema)
+
+        const modelSchema = this.generator.getSchemaForSymbol(modelName)
+        ajv.validateSchema(modelSchema)
+        if (ajv.errors) {
+            throw new Error(`Schema for class ${modelName} was not valid!`)
+        }
+        const maskedSchema = WebHookModelValidator._maskSchemaTypes(modelSchema)
+        ajv.validateSchema(maskedSchema)
+        if (ajv.errors) {
+            throw new Error(`Masked schema for class ${modelName} was not valid!`)
+        }
+        this.fieldsValidators[modelName] = ajv.compile(maskedSchema)
+    }
+
+    /**
+     * Validates specified model filters
+     * @param {string} modelName
+     * @param {JSON} filters
+     * @returns {{isValid: boolean, errors: *[]}}
+     */
+    validateFilters (modelName, filters) {
+        if (!this.models.includes(modelName)) {
+            throw new Error(`Unregistered model name: ${modelName}`)
+        }
+        const validator = this.filterValidators[modelName]
+        const isValid = validator(filters)
```

Comment:
In the case `this.filterValidators[modelName]` is undefined or not callable, we get an exception

Reply:
We assume that function input are not falsey (already preprocessed by validateInput or something like that)